radio/src/targets/common/arm/stm32/audio_dac_driver.cpp

```diff
@@ -100,28 +104,48 @@ void audioMute()
   }
   else if (now - audioQueue.lastAudioPlayTime > AUDIO_MUTE_DELAY / 10) {
     // delay expired, we may mute
-    GPIO_SetBits(AUDIO_MUTE_GPIO, AUDIO_MUTE_GPIO_PIN);
+#if defined(INVERTED_MUTE_PIN)
+  GPIO_ResetBits(AUDIO_MUTE_GPIO, AUDIO_MUTE_GPIO_PIN);
+#else
+  GPIO_SetBits(AUDIO_MUTE_GPIO, AUDIO_MUTE_GPIO_PIN);
+#endif
   }
 #else
   // mute
+#if defined(INVERTED_MUTE_PIN)
+  GPIO_ResetBits(AUDIO_MUTE_GPIO, AUDIO_MUTE_GPIO_PIN);
+#else
   GPIO_SetBits(AUDIO_MUTE_GPIO, AUDIO_MUTE_GPIO_PIN);
 #endif
+#endif
 }
 
 void audioUnmute()
 {
 #if defined(AUDIO_UNMUTE_DELAY)
   // if muted
+#if defined(INVERTED_MUTE_PIN)
+  if (!GPIO_ReadOutputDataBit(AUDIO_MUTE_GPIO, AUDIO_MUTE_GPIO_PIN)) {
+    // ..un-mute
+    GPIO_SetBits(AUDIO_MUTE_GPIO, AUDIO_MUTE_GPIO_PIN);
+    RTOS_WAIT_MS(AUDIO_UNMUTE_DELAY);
+  }
+#else
   if (GPIO_ReadOutputDataBit(AUDIO_MUTE_GPIO, AUDIO_MUTE_GPIO_PIN)) {
     // ..un-mute
     GPIO_ResetBits(AUDIO_MUTE_GPIO, AUDIO_MUTE_GPIO_PIN);
     RTOS_WAIT_MS(AUDIO_UNMUTE_DELAY);
   }
+#endif
   // reset the mute delay
   audioQueue.lastAudioPlayTime = 0;
+#else
+#if defined(INVERTED_MUTE_PIN)
+  GPIO_SetBits(AUDIO_MUTE_GPIO, AUDIO_MUTE_GPIO_PIN);
 #else
   GPIO_ResetBits(AUDIO_MUTE_GPIO, AUDIO_MUTE_GPIO_PIN);
 #endif
+#endif
```

Comment:
Maybe it would make sense to have one function setting the mute pin correctly instead of having four places where the define is checked.

Reply:
Can I ask, how would that function look like?

Reply:
something like
```
static inline void mute(bool mute)
{
#if defined(INVERTED_MUTE_PIN)
  if(mute)
    GPIO_ResetBits(AUDIO_MUTE_GPIO, AUDIO_MUTE_GPIO_PIN);
  else
    GPIO_SetBits(AUDIO_MUTE_GPIO, AUDIO_MUTE_GPIO_PIN);
#else
  if(mute)
    GPIO_SetBits(AUDIO_MUTE_GPIO, AUDIO_MUTE_GPIO_PIN);
  else
    GPIO_ResetBits(AUDIO_MUTE_GPIO, AUDIO_MUTE_GPIO_PIN);
#endif
}
```

the audioUnmute with delay could the be reduced to 
```
#if defined(INVERTED_MUTE_PIN)
if (!GPIO_ReadOutputDataBit(AUDIO_MUTE_GPIO, AUDIO_MUTE_GPIO_PIN)) {
#else
if (!GPIO_ReadOutputDataBit(AUDIO_MUTE_GPIO, AUDIO_MUTE_GPIO_PIN)) {
#endif
  mute(false);
  RTOS_WAIT_MS(AUDIO_UNMUTE_DELAY);
}
```
and all other places just call this one function with the correct parameter. The compiler should then be able to optimize everything away, except for the one call to set or reset the pin.
@raphaelcoeffic ?
@pfeerick ?

Reply:
mux better, thx